packages/client/hooks/useMedia.ts

```diff
@@ -0,0 +1,25 @@
+import {ProducerState, ConsumerState} from '../utils/mediaRoom/reducerMediaRoom'
+import {useEffect} from 'react'
+
+interface useMediaSignature {
+  mediaRef: React.RefObject<HTMLVideoElement> | React.RefObject<HTMLAudioElement>
+  mediaSource: ProducerState | ConsumerState | undefined
+}
+
+const useMedia = ({mediaRef, mediaSource}: useMediaSignature) => {
+  useEffect(() => {
+    if (mediaSource?.track) {
+      const stream = new MediaStream()
+      stream.addTrack(mediaSource.track)
+      const el = mediaRef.current!
+      if (el.srcObject !== stream) el.srcObject = stream! // conditional is required to remove flickering video on update
+    }
+  })
```

Comment:
may be worth passing in a list of deps here so we aren't creating a new mediastream on every re-render

Reply:
that sounds like a good idea!